### DISCUSSION TITLE
Rain forecast

### DISCUSSION THREAD

**Discussion by BuyMyMojo** [ttytm/wthrr-the-weathercrab#29] [Ideas]
This might be part of that `Allow to show more / customized meteorological data` part of the outlook but being able to see the chance of rain through the day like how google shows it would be nice.

![Google rain forecast example](https://s3.buymymojo.net/ShareX/2022/08/19/07/chrome_NHIxn1SOzi.png)

**Comment by ttytm** [comment]
Great suggestions. I will definetly work on something like that. Thank you!

**Comment by BuyMyMojo** [comment]
Looking into it and the Open-Metro API does seem to have a [rain forecast API](https://open-meteo.com/en/docs#latitude=-37.86&longitude=144.68&hourly=rain) which results in two lists:
time
mm of rain fall

There is a seperate showers call which outputs the same data format but idk what the difference between rain and showers are.

There is also a seperate snowfall one but that uses cm not mm.

One final note which is unrelated but there is a freezing height graph that shows at what height water freezes which is just kinda fun.

**Comment by ttytm** [comment]
Hey @BuyMyMojo,

Thanks for investigating and sorry about the delayed response. I'm in a very busy schedule until next week, unfortunately.

After you open this issue, something like the "chance of rain for a day" felt like a very common and useful thing to have in a weather report. I wondered why I hadn't something like that already integrated. So, for now, I thought of adding a little indicator that shows the probability of rain for a day. This would be quickly and fairly easy to integrate. And for the longer term, I would add something like it is done on the screenshot you have posted. 

But it turned out - as you also have found out - the API doesn't make something like a "probability of precipitation" available. 

Edit: There is an open issue on extening the API to integrate [Precipitation probability](https://github.com/open-meteo/open-meteo/issues/85).

> Looking into it and the Open-Metro API does seem to have a [rain forecast API](https://open-meteo.com/en/docs#latitude=-37.86&longitude=144.68&hourly=rain) which results in two lists:
time
mm of rain fall

Right, this could be a good substitute I think.

> One final note which is unrelated but there is a freezing height graph that shows at what height water freezes which is just kinda fun.

Currently, it's summer at my place, until winter (due to limited time) hopefully I can integrate those nicer winter related stats 😊.

**Comment by ttytm** [comment]
@BuyMyMojo  

Something that comes close was added with #18 .

Until a chance of precipitation is available via the API the best we could do is to use the amount of precipitation in millimeters. 

I'm releasing new version containing this new feature soon.
